apps/system/utils/utils_heapinfo.c

```diff
@@ -218,14 +221,33 @@ static void heapinfo_show_taskinfo(struct mm_heap_s *heap)
 #endif
 	printf("-------|-----------|-----------|----------\n");
 
+#if CONFIG_MM_NHEAPS > 1
+	int tcb_pid[CONFIG_MAX_TASKS];
+	for (tcb_idx = 0; tcb_idx < CONFIG_MAX_TASKS; tcb_idx++) {
+		tcb_pid[tcb_idx] = HEAPINFO_INIT_INFO;
+	}
+#endif
+
 	for (heap_idx = 0; heap_idx < CONFIG_MM_NHEAPS; heap_idx++) {
 		for (tcb_idx = 0; tcb_idx < CONFIG_MAX_TASKS; tcb_idx++) {
 			if (heap[heap_idx].alloc_list[tcb_idx].pid != HEAPINFO_INIT_INFO) {
+#if CONFIG_MM_NHEAPS > 1
```

Comment:
plz check the possibility of using tcb_pid as a flag only, for simple logic of initialize it.

Reply:
fixed, thanks